config/config_test.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var testDir string = "./testdata"
+
+func TestReadConfig(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		testConfig := Config{
+			APIURL:   "api-url",
+			StoreURL: "store-api-url",
+			Token:    "dummy_token",
+			Launcher: Launcher{
+				Version: "latest",
+				Image:   "screwdrivercd/launcher",
+			},
+		}
+
+		actual, err := ReadConfig(testDir)
+
+		assert.Nil(t, err)
+		assert.Equal(t, testConfig, actual)
+	})
+
+	t.Run("failure by no entry", func(t *testing.T) {
+		emptyConfig := Config{}
+		actual, err := ReadConfig("./not-exist")
+		assert.NotNil(t, err, fmt.Sprintf("There is no error when reading config fails"))
+
+		assert.Equal(t, emptyConfig, actual)
+
+		msg := err.Error()
+		assert.Equal(t, 0, strings.Index(msg, "failed to read config file: "), fmt.Sprintf("expected error is `failed to read config file: ...`, actual: `%v`", msg))
+	})
+
+	t.Run("failure by invalid yaml", func(t *testing.T) {
+		invalidConfigFile := []byte("key; 1")
+		tmpdir, err := ioutil.TempDir("", "sdlocal")
+
+		assert.Nil(t, err, fmt.Sprintf("failed to create temporary directory: %v", err))
+
+		defer os.RemoveAll(tmpdir)
+
+		err = ioutil.WriteFile(path.Join(tmpdir, "config"), invalidConfigFile, 0666)
```

Comment:
How about adding an invalid file instead of creating an invalid file in codes? It makes codes smaller.

Reply:
I think so.
but it is difficult because config file name is fixed to `config`.
I will fix it if I need to use other test files in the future.

Reply:
How about to dig directories like `testdata/success` and `testdata/invalid`?

Reply:
I reconsidered.
fixed

Reply:
Instead of digging directories, we decided to pass them by argument.